File: roktracker/utils/types/scan_preset.py
from dataclasses import dataclass
from pydantic import BaseModel
from typing import List, Optional
from enum import StrEnum


class ScanItems(StrEnum):
    ID = "ID"
    NAME = "Name"
    POWER = "Power"
    KILLPOINTS = "Killpoints"
    ALLIANCE = "Alliance"
    T1_KILLS = "T1 Kills"
    T2_KILLS = "T2 Kills"
    T3_KILLS = "T3 Kills"
    T4_KILLS = "T4 Kills"
    T5_KILLS = "T5 Kills"
    RANGED = "Ranged"
    DEATHS = "Deaths"
    ASSISTANCE = "Assistance"
    GATHERED = "Gathered"
    HELPS = "Helps"


class ScanPreset(BaseModel):
    name: str
    selections: List[ScanItems]


@dataclass
class ScanOptions:
    id: bool = True
    name: bool = True
    power: bool = True
    killpoints: bool = True
    alliance: bool = True
    t1_kills: bool = True
    t2_kills: bool = True
    t3_kills: bool = True
    t4_kills: bool = True
    t5_kills: bool = True
    ranged: bool = True
    deaths: bool = True
    assistance: bool = True
    gathered: bool = True
    helps: bool = True

File: gui_frontend/src/schema/ScanPreset.ts
import { z } from "zod";

export const ScanPresetSchema = z.object({
  name: z.string(),
  selections: z.array(
    z.enum([
      "ID",
      "Name",
      "Power",
      "Killpoints",
      "Alliance",
      "T1 Kills",
      "T2 Kills",
      "T3 Kills",
      "T4 Kills",
      "T5 Kills",
      "Ranged",
      "Deaths",
      "Assistance",
      "Gathered",
      "Helps",
    ]),
  ),
});
export type ScanPreset = z.infer<typeof ScanPresetSchema>;
